README.md
# openalpr-android
[![Release](https://jitpack.io/v/SandroMachado/openalpr-android.svg)](https://jitpack.io/#SandroMachado/openalpr-android)

OpenALPR is an open source Automatic License Plate Recognition library written in C++ with bindings in C#, Java, Node.js, and Python. This project ports this library to Android. You can find the demo application `apk` at the [releases](https://github.com/SandroMachado/openalpr-android/releases) tab.

![Screenshot](images/screenshot.png "Main Activity Sample application")

# Gradle Dependency

## Repository

First, add the following to your app's `build.gradle` file:

```Gradle
repositories {
    maven { url "https://jitpack.io" }
}
```

Them include the openalpr-android dependency:

```gradle
dependencies {

    // ... other dependencies here.    	
    compile 'com.github.SandroMachado:openalpr-android:1.1.2'
}
```

# Usage

## Code

Copy the [OpenALPR configuration file](./openalpr.conf) to your android project assets directory `/main/assets/runtime_data/openalpr.conf`, open it and update the `runtime_dir` to your project directory (for instance, for the sample project the directory is: `runtime_dir = /data/data/com.sandro.openalprsample/runtime_data`). After that just follow the code example bellow. To see a full example check the [sample application](./Sample/OpenALPRSample/app/src/main/java/com/sandro/openalprsample/MainActivity.java).

```Java

static final String ANDROID_DATA_DIR = "/data/data/com.sandro.openalprsample";

final String openAlprConfFile = ANDROID_DATA_DIR + File.separatorChar + "runtime_data" + File.separatorChar + "openalpr.conf";

String result = OpenALPR.Factory.create(MainActivity.this, ANDROID_DATA_DIR).recognizeWithCountryRegionNConfig("us", "", image.getAbsolutePath(), openAlprConfFile, 10);
```

## Interface

```Java
/*
 Method interface.
*/

/**
 * Recognizes the licence plate.
 *
 * @param country        - Country code to identify (either us for USA or eu for Europe). Default=us.
 * @param region         -  Attempt to match the plate number against a region template (e.g., md for Maryland, ca for California).
 * @param imgFilePath    - Image containing the license plate.
 * @param configFilePath - Config file path (default /etc/openalpr/openalpr.conf)
 * @param topN           - Max number of possible plate numbers to return(default 10)
 *
 * @return - JSON string of results
 */

public String recognizeWithCountryRegionNConfig(String country, String region, String configFilePath, String imgFilePath, int topN);

```
# Sample Application

The repository also includes a [sample application](./Sample/OpenALPRSample) that can be tested with Android Studio.

![Screencast](images/screencast.gif "Main Activity Sample application screencast")

# Credits

 - [OpenALPR](https://github.com/openalpr/openalpr) Parent Project
 - [OpenAlprDroidApp](https://github.com/sujaybhowmick/OpenAlprDroidApp) for the compiled sources and sample that helped port the project to an android library


CHANGELOG.md
# Changelog

## [1.1.2](https://github.com/SandroMachado/openalpr-android/tree/1.1.2) (2017-03-09)
[Full Changelog](https://github.com/SandroMachado/openalpr-android/compare/1.1.1...1.1.2)

**Closed issues:**

- Error initializing OpenALPR [\#42](https://github.com/SandroMachado/openalpr-android/issues/42)
- I'm getting an error [\#40](https://github.com/SandroMachado/openalpr-android/issues/40)
- Can I specify country settings in openalpr.config [\#39](https://github.com/SandroMachado/openalpr-android/issues/39)
- Error :\( [\#38](https://github.com/SandroMachado/openalpr-android/issues/38)
- I don't know why it doesn't work... [\#36](https://github.com/SandroMachado/openalpr-android/issues/36)
- Upgrade the dependency `Libpng library` to allow submission to `Play Store` [\#35](https://github.com/SandroMachado/openalpr-android/issues/35)
- Openalpr is Apache 2.0 but original openalpr is AGPL 3.0. Which license affects me? [\#32](https://github.com/SandroMachado/openalpr-android/issues/32)
- issue in changing package name  [\#29](https://github.com/SandroMachado/openalpr-android/issues/29)
- seems couldn't recognize words? [\#21](https://github.com/SandroMachado/openalpr-android/issues/21)
- How to pass multiple country code? [\#20](https://github.com/SandroMachado/openalpr-android/issues/20)
- Pass image as byte array not as file path [\#18](https://github.com/SandroMachado/openalpr-android/issues/18)
- No module found [\#17](https://github.com/SandroMachado/openalpr-android/issues/17)
- Failed to resolve: com.github.SandroMachado:openalpr-android:1.1.0 [\#16](https://github.com/SandroMachado/openalpr-android/issues/16)
- libopenalpr-native.so & libopencv\_java.so [\#15](https://github.com/SandroMachado/openalpr-android/issues/15)
- library "libopencv\_java.so" not found [\#14](https://github.com/SandroMachado/openalpr-android/issues/14)
- library "libopencv\_java.so" not found [\#13](https://github.com/SandroMachado/openalpr-android/issues/13)

**Merged pull requests:**

- Update libpng library [\#44](https://github.com/SandroMachado/openalpr-android/pull/44) ([SandroMachado](https://github.com/SandroMachado))
- Multiple code quality fix-2 [\#28](https://github.com/SandroMachado/openalpr-android/pull/28) ([faisal-hameed](https://github.com/faisal-hameed))
- Code quality fix - Utility classes should not have public constructors. [\#26](https://github.com/SandroMachado/openalpr-android/pull/26) ([faisal-hameed](https://github.com/faisal-hameed))
- Code quality fix - Method names should comply with a naming convention. [\#25](https://github.com/SandroMachado/openalpr-android/pull/25) ([faisal-hameed](https://github.com/faisal-hameed))
- Code quality fix - Declarations should use Java collection interfaces such as "List" rather than specific implementation. [\#24](https://github.com/SandroMachado/openalpr-android/pull/24) ([faisal-hameed](https://github.com/faisal-hameed))
- Update README.md [\#23](https://github.com/SandroMachado/openalpr-android/pull/23) ([gas83](https://github.com/gas83))
- Update README.md [\#22](https://github.com/SandroMachado/openalpr-android/pull/22) ([gas83](https://github.com/gas83))

## [1.1.1](https://github.com/SandroMachado/openalpr-android/tree/1.1.1) (2016-01-14)
[Full Changelog](https://github.com/SandroMachado/openalpr-android/compare/1.1.0...1.1.1)

**Merged pull requests:**

- Update JitPack badge [\#11](https://github.com/SandroMachado/openalpr-android/pull/11) ([SandroMachado](https://github.com/SandroMachado))
- Add portuguese licence plate key points [\#10](https://github.com/SandroMachado/openalpr-android/pull/10) ([SandroMachado](https://github.com/SandroMachado))
- \#4 Added new keypoint folder for Europe with Polish LP placeholders [\#9](https://github.com/SandroMachado/openalpr-android/pull/9) ([JerzyPuchalski](https://github.com/JerzyPuchalski))
- Created a folder on /sdcard/ for all images captured [\#8](https://github.com/SandroMachado/openalpr-android/pull/8) ([ZKjellberg](https://github.com/ZKjellberg))

## [1.1.0](https://github.com/SandroMachado/openalpr-android/tree/1.1.0) (2016-01-06)
[Full Changelog](https://github.com/SandroMachado/openalpr-android/compare/1.0.1...1.1.0)

**Fixed bugs:**

- \[Sample Application\] State Lost on Rotation [\#6](https://github.com/SandroMachado/openalpr-android/issues/6)
- \[Sample Application\] Crash on Invalid Submission [\#5](https://github.com/SandroMachado/openalpr-android/issues/5)

**Closed issues:**

- Crash during EU recognizing [\#4](https://github.com/SandroMachado/openalpr-android/issues/4)
- Crash caused by missing libopenalpr-native.so [\#1](https://github.com/SandroMachado/openalpr-android/issues/1)

**Merged pull requests:**

- Long list of improvements [\#7](https://github.com/SandroMachado/openalpr-android/pull/7) ([ZKjellberg](https://github.com/ZKjellberg))
- IntelliJ Files Cleanup [\#3](https://github.com/SandroMachado/openalpr-android/pull/3) ([ZKjellberg](https://github.com/ZKjellberg))

## [1.0.1](https://github.com/SandroMachado/openalpr-android/tree/1.0.1) (2015-12-17)
[Full Changelog](https://github.com/SandroMachado/openalpr-android/compare/1.0.0...1.0.1)

**Merged pull requests:**

- Implemented permission check [\#2](https://github.com/SandroMachado/openalpr-android/pull/2) ([ZKjellberg](https://github.com/ZKjellberg))

## [1.0.0](https://github.com/SandroMachado/openalpr-android/tree/1.0.0) (2015-12-13)


\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

app/src/java/org/openalpr/app/AppConstants.java
package org.openalpr.app;

/**
 * Created by sujay on 28/11/14.
 */
public interface AppConstants {
    static final String BITMAP_STORAGE_KEY = "viewbitmap";
    static final String IMAGEVIEW_VISIBILITY_STORAGE_KEY = "imageviewvisibility";

    static final int REQUEST_IMAGE_CAPTURE = 1;
    static final String RUNTIME_DATA_DIR_ASSET = "runtime_data";
    static final String ANDROID_DATA_DIR = "/data/data/org.openalpr.app";
    static final String OPENALPR_CONF_FILE = "openalpr.conf";
    static final String PREF_INSTALLED_KEY = "installed";
    static final String JPEG_FILE_PREFIX = "IMG_";
    static final String JPEG_FILE_SUFFIX = ".jpg";
    static final String ALPR_FRAGMENT_TAG = "alpr";
    static final String ALPR_ARGS = "alprargs";

    static final String DLG_MESSAGE = "dlgMessage";

    static final String JSON_RESULT_ARRAY_NAME = "results";
}
